packages/react-dom/src/test-utils/ReactTestUtils.js

```diff
@@ -390,58 +444,70 @@ const ReactTestUtils = {
 
   Simulate: null,
   SimulateNative: {},
-
-  act(callback: () => void): Thenable {
+  act(callback: () => void | Promise<void>) {
     if (actContainerElement === null) {
-      // warn if we can't actually create the stub element
       if (__DEV__) {
-        warningWithoutStack(
-          typeof document !== 'undefined' &&
-            document !== null &&
-            typeof document.createElement === 'function',
-          'It looks like you called TestUtils.act(...) in a non-browser environment. ' +
-            "If you're using TestRenderer for your tests, you should call " +
-            'TestRenderer.act(...) instead of TestUtils.act(...).',
-        );
+        // warn if we're trying to use this in something like node (without jsdom)
+        if (didWarnAboutActInNodejs === false) {
+          didWarnAboutActInNodejs = true;
+          warningWithoutStack(
+            typeof document !== 'undefined' && document !== null,
+            'It looks like you called ReactTestUtils.act(...) in a non-browser environment. ' +
+              "If you're using TestRenderer for your tests, you should call " +
+              'ReactTestRenderer.act(...) instead of ReactTestUtils.act(...).',
+          );
+        }
       }
-      // then make it
+      // now make the stub element
       actContainerElement = document.createElement('div');
     }
-
-    const result = ReactDOM.unstable_batchedUpdates(callback);
     // note: keep these warning messages in sync with
     // createReactNoop.js and ReactTestRenderer.js
-    if (__DEV__) {
-      if (result !== undefined) {
-        let addendum;
-        if (result !== null && typeof result.then === 'function') {
-          addendum =
-            '\n\nIt looks like you wrote ReactTestUtils.act(async () => ...), ' +
-            'or returned a Promise from the callback passed to it. ' +
-            'Putting asynchronous logic inside ReactTestUtils.act(...) is not supported.\n';
-        } else {
-          addendum = ' You returned: ' + result;
-        }
-        warningWithoutStack(
-          false,
-          'The callback passed to ReactTestUtils.act(...) function must not return anything.%s',
-          addendum,
-        );
+    const result = actedUpdates(callback, runCallbackUntilPredicateFails);
+    if (
+      result !== null &&
+      typeof result === 'object' &&
+      typeof result.then === 'function'
+    ) {
+      let called = false;
+      if (__DEV__) {
+        // $FlowFixMe - if we've reached this far, we're certain Promise is available
```

Comment:
This is in a dev block, anyway, and Promise will almost certainly always be defined, so it can use an existence check. The bar for using $FlowFixMe needs to be really high and I don't think it's necessary in this case.

Alternatively we should decide that Promise is a required global and update our ESLint config accordingly.

Reply:
the problem with an existence check, is that flow infers that Promise might not exist as a global, and it throws weird errors. sent you a link privately to internal post on the same.

Reply:
That error doesn't seem to relevant to me. It says Promise is not a polymorphic type, which is more a problem with Flow's built-in type defs. But that shouldn't matter here because you don't need to type the resolved value; you only need to call `Promise.resolve().then`:

https://flow.org/try/#0AoJw9gtglgzgpgOhHGYA2A3OAKAlAgFwAs4A7PIA

Reply:
It poisons any function with a Promise in its type annotation. I'll keep trying to fix this, but fair warning I already tried a bit and kept hitting walls. 
I would heartily +1 updating the eslint config. It seems reasonable at this point.

Reply:
You don't need to use `Promise` in the types of any of these functions though. You use `Thenable<T>`.

Reply:
Alright, I must have been doing something wrong earlier. I "fixed" this.